app/controlplane/internal/service/organization.go

```diff
@@ -59,6 +59,25 @@ func (s *OrganizationService) Create(ctx context.Context, req *pb.OrganizationSe
 	return &pb.OrganizationServiceCreateResponse{Result: bizOrgToPb(org)}, nil
 }
 
+func (s *OrganizationService) Update(ctx context.Context, req *pb.OrganizationServiceUpdateRequest) (*pb.OrganizationServiceUpdateResponse, error) {
+	currentUser, err := requireCurrentUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	org, err := s.orgUC.FindByName(ctx, req.Name)
+	if err != nil {
+		return nil, handleUseCaseErr(err, s.log)
```

Comment:
Before retrieving any entity, I would check that the user has access to it. I know it's done in `orgUC.Update`, but there it's too late in my opinion (not failing here would give clues that the org exist, even if it returns a 404 in that method).

Reply:
Updated, thanks!